src/contact/parallel/bvh_contact_manager.cc

```diff
@@ -19,23 +19,20 @@ namespace nimble {
         query_tree_local( tree, elb, [&_a, &_b, &elb, &res, this]( std::size_t _i ) {
           const auto &face = _a.elements[_i];
           const auto &node = elb;
-          ContactEntity::vertex closest_point;
           ContactManager::PROJECTION_TYPE proj_type;
-          contact_manager->ClosestPointProjectionSingle( node, face, &closest_point, &proj_type,
-              1.0e-16 );
-
-          // Compute distance
-          double dx = node.coord_1_x_ - closest_point[0];
-          double dy = node.coord_1_y_ - closest_point[1];
-          double dz = node.coord_1_z_ - closest_point[2];
-          double gap = std::sqrt( dx * dx + dy * dy + dz * dz );
-
-          if ( gap <= 0.15 * ( node.char_len_ + face.char_len_ ) ) {
-            auto &entry = res.emplace_back();
-            entry.first_global_id = face.contact_entity_global_id();
-            entry.second_global_id = node.contact_entity_global_id();
-            entry.gap = gap;
-          }
+          ContactEntity::vertex closest_point;
+          double gap;
+          double normal[3];
+          contact_manager->ClosestPointProjectionSingle( node, face, 
+            &proj_type, &closest_point, gap, &normal);
+          if ( proj_type != ContactManager::PROJECTION_TYPE::UNKNOWN ) {
+            if ( gap <= 0.15 * ( node.char_len_ + face.char_len_ ) ) {
```

Comment:
@nmm0 @rjones-sandia 
Just a thought... In order to keep some quantities(such as tolerances) identical for BVH and ArborX, should we generate the right-hand side from the contact entity face?

Reply:
yes I think we should keep the two broad searches consistent as possible but do you mean
if ( gap <= 0.15 * (  face.char_len_ ) ) {
i.e. skip the node size since it may not be consistent between the two searches?
sorry I don't quite understand the suggestion

Reply:
I meant something like ```if (gap <= face.gapThreshold())``` and in the `ContactEntity`, add the function `double gapThreshold() { return 0.15 * char_len_; }` 
Maybe it is an overkill...

Reply:
I don't think it's overkill; I'm not sure where exactly the 0.15 comes from in the code -- but if it's probably better to consolidate that calculation so if we ever find we need to change it we could easily do so by changing the gapThreshold. But maybe name it something like `getEntityExpansionDistance`? Because it's also used when generating the bounding boxes.

Reply:
yeah I agree but I still don't fully understand. Is the "char_len_" or whatever gives us a sense of size of the facet not consistent across ArborX vs BVH implementations? 

I think the "0.15" should stay in one function (the projection for now). I like to put it in the call args but define it 
e.g.. function(..... tol = 0.15) so I know it's an arbitrary constant but there is a common guess

Reply:
I do not know much about contact and contact algorithms. 
Are the two lengths (the bounding box inflation length and the gap _threshold_ length --- i.e. the closeness of a node in the normal direction to a face) related?
I see in the code that we use the same value: 0.15 times a face characteristic length. 
But, to my naive eyes, these two aspects appear independent. 
--- 
I was suggesting in the class ContactEntity (rather than in the projection routine) as the gap _threshold_ length determines whether a contact force is enforced or not (--- is my understanding correct?)

Reply:
It seems like we agree on keeping the empirical constant(s) (like 0.15) in one unique place. Maybe the best place remains to be decided.